### DISCUSSION TITLE
What is the purpose of `custom/configs/overrides.lua`

### DISCUSSION THREAD

**Discussion by bhuynhdev** [NvChad/NvChad#2653] [Q&A] [answered]
I have been reading through how to do custom configuration with NVChad through the `custom` directory. One thing I am confused about is that there seems to be two ways to change plugin's config:
1. By adding/removing/modifying the `custom/plugins.lua` file directly

Or

2. Changing inside the `configs/overrides.lua` file --> This seems to apply only to `mason` and `nvimtree`

It seems like the preferred way is option (1). For example, to override the `nvim-cmp` mappings, [siduck suggested to use the `custom/plugins.lua` file instead of the `overrides.lua` file](https://github.com/NvChad/NvChad/discussions/2605)

I wish to ask: Is there any differences between the two options? What is the purpose of this `overrides` file? Why does it only apply to `mason` and `nvimtree`? And when should/shouldn't I use it?

Thanks a lot for your help.

**Comment by siduck** [comment, marked as answer]
that file path is just an example, you can name it anything, and its not mandatory, if you dont use it  then you'll have to write all the plugin's opts table within the same file i.e `custom.plugins` , so instead of that we define the opts tables in a separate file which is ` custom/configs/overrides` and then we import that in `custom.plugins`

you shouldnt use it when you think the plugin's opts has a module defined like the cmp one, so we make opts a function instead so the module wont be called at startup
